holidays/countries/yemen.py

```diff
@@ -0,0 +1,138 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars import _CustomIslamicHolidays
+from holidays.calendars.gregorian import (
+    MAR,
+    APR,
+    MAY,
+    JUN,
+    JUL,
+    AUG,
+    SEP,
+    OCT,
+)
+from holidays.groups import InternationalHolidays, IslamicHolidays
+from holidays.holiday_base import HolidayBase
+
+
+class Yemen(HolidayBase, InternationalHolidays, IslamicHolidays):
+    """Yemen holidays.
+
+    References:
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Yemen>
+        * <https://yemenembassy.it/festivities/>
+    """
+
+    country = "YE"
+    default_language = "ar"
+    # %s (estimated).
+    estimated_label = tr("%s (مُقدَّر)")
+    start_year = 1990
+    supported_languages = ("ar", "en_US")
+
+    def __init__(self, islamic_show_estimated: bool = True, *args, **kwargs):
+        """
+        Args:
+            islamic_show_estimated:
+                Whether to add "estimated" label to Islamic holidays name
+                if holiday date is estimated.
+        """
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(
+            self, cls=YemenIslamicHolidays, show_estimated=islamic_show_estimated
+        )
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # Labor Day.
+        self._add_labor_day(tr("عيد العمال"))
+
+        # Unity Day.
+        self._add_holiday_may_22(tr("اليوم الوطني للجمهورية اليمنية"))
+
+        # Revolution Day.
+        self._add_holiday_sep_26(tr("ثورة 26 سبتمبر المجيدة"))
+
+        # Liberation Day.
+        self._add_holiday_oct_14(tr("ثورة 14 أكتوبر المجيدة"))
+
+        # Independence Day.
+        self._add_holiday_nov_30(tr("عيد الجلاء"))
+
+        # Hijri New Year.
+        self._add_islamic_new_year_day(tr("عيد راس السنة الهجرية"))
+
+        # Prophet's Birthday.
+        self._add_mawlid_day(tr("المولد النبوي"))
+
+        # Eid al-Fitr.
+        name = tr("عيد الفطر")
+        self._add_eid_al_fitr_day(name)
+        self._add_eid_al_fitr_day_two(name)
+        self._add_eid_al_fitr_day_three(name)
+
+        # Eid al-Adha.
+        name = tr("عيد الأضحى")
+        self._add_eid_al_adha_day(name)
```

Comment:
> The occasion of Eid Al-Adha starts on the 9th of Dhu al-Hijjah and ends on the fourth day of Eid.

I don't think we got `_add_eid_al_adha_eve` at the moment for `group/islamic.py` but feel free to add them.
```suggestion
        name = tr("عيد الأضحى")
        self._add_eid_al_adha_eve(name)
        self._add_eid_al_adha_day(name)
```

Reply:
9 Dhu al-Hijjah is actually Arafah Day, and we have method for it.